### PR TITLE
Clean tags #

### LINKED ISSUE #112: Delete hashtag when adding in tag
When adding tags in the database, sometimes they are preceded by a hashtag, making them a separate entry. We should remove it to avoid cluttering the database.

### User Story

As a developer, I want to avoid cluttering the database with uncleaned data, so that queries are more efficient and there's less errors when retrieving data.

### PR DESCRIPTION
**Pull request by e-lemma**
I have improved the transform script to remove any leading hashtags to prevent identical tags like "#pop" and "pop" being considered different tags. This change will result in more accurate and complete data.
Resolves #112.

### CODE REVIEW

File: pipeline/test_transform.py [hunk 1]
```diff
@@ -150,24 +150,28 @@ def test_transform_sales_data_valid():
 
 
 def test_clean_tags_normal_case():
-    tags = [" TagOne ", "TAGTWO", "tagthree "]
+    """Tests that tags are properly cleaned in normal cases."""
+    tags = [" TagOne ", "##TAGTWO", "#tagthree "]
```
Comment (Zhi-704) [inline pipeline/test_transform.py:154]: Good test for edge cases

File: pipeline/test_transform.py [hunk 2]
```diff
@@ -150,24 +150,28 @@ def test_transform_sales_data_valid():
...
 
 def test_clean_tags_empty_list():
+    """Tests that an empty list is handled correctly."""
     tags = []
     expected = []
     assert clean_tags(tags) == expected
 
 
 def test_clean_tags_none():
+    """Tests that 'None' is handled correctly."""
```
Comment (Zhi-704) [inline pipeline/test_transform.py:167]: Haha missed the docstrings in the last pull request. Good job adding them in

File: pipeline/transform.py [hunk 3]
```diff
@@ -59,10 +59,10 @@ def clean_data(sales_data: dict) -> dict:
 
 
 def clean_tags(tags: list[str]) -> list[str]:
-    """Removes leading/trailing whitespace and 'lowercases' a list of tags, returning it."""
+    """Removes leading/trailing whitespace, leading hashtags, and 'lowercases' a list of tags, returning it."""
     if tags is None:
         return []
-    return [tag.strip().lower() for tag in tags]
+    return [tag.lstrip("#").strip().lower() for tag in tags]
```
Comment (Zhi-704) [inline pipeline/transform.py:65]: Seems good

**Review by Zhi-704 — APPROVED** [review]
